Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -39,6 +41,11 @@ static std::thread s_prefetcher;
 
 static const std::string s_format_prefix = "tex1_";
 
+HiresTexture::Level::Level()
+	: data(nullptr, SOIL_free_image_data)
```

Comment:
It won't. But that's still a dangerous thing to do (have a null deleter). If you reset anything that isn't a nullptr into it, you'll crash the program from a null pointer dereference.

e.g. Hypothetically doing.
```cpp
SOILPointer ptr1(new unsigned char, SOIL_free_image_data);
SOILPointer ptr2(nullptr, nullptr);
ptr2.reset(ptr1.release());
```

will cause a crash, the deleter isn't transferred over. deleters should *always* have a valid state to avoid that (for example the default unique_ptr's deleter is `std::default_deleter<T>`, not nullptr)